bigchaindb/tendermint/lib.py

```diff
@@ -347,12 +342,14 @@ def validate_transaction(self, tx, current_transactions=[]):
             except ValidationError as e:
                 logger.warning('Invalid transaction (%s): %s', type(e).__name__, e)
                 return False
+        return transaction.validate(self, current_transactions)
+
+    def is_valid_transaction(self, tx, current_transactions=[]):
```

Comment:
Why can'y we always return True/False? Then also docstring is not needed since the function is self-explanatory.

Reply:
It is used in the code in [an assignment](https://github.com/bigchaindb/bigchaindb/pull/2270/files#diff-c45065100e5e308772764cdf833def41R83).

Reply:
But the transaction hopefully does not change during validation. So that code can just use the object it passes to the function.

Reply:
@vrde @ldmberman I think the concern raised is valid but I feel the context of this PR is getting too broad. Fixing the return value for `is_valid_transaction` would also require chainging the response values in the ABCI proxy app. I think this refactoring should be done in a seperate PR?

Reply:
If it's only about the place @vrde pointed to, I would fix it now cause it's very simple.

Reply:
@ldmberman Could you please explain the fix you have in mind?

Reply:
Instead of:
```
transaction = self.bigchaindb.is_valid_transaction(
             decode_transaction(raw_transaction), self.block_transactions)
if not transaction:
    ...
```
do:
```
transaction = decode_transaction(raw_transaction)
if not self.bigchaindb.is_valid_transaction(transaction, self.block_transactions):
    ...
```

Reply:
Hmm, the return value in the `transaction` variable is an object and not a dict. The function call `store_bulk_transactions` in the `commit` API expects an object not a dict. Also the corresponding tests for `store_bulk_transactions` will have to be fixed.

Reply:
Ah, I see! Can we then do `Transaction.from_dict` there?

Reply:
@ldmberman Yup. But as soon as we make that change we will need to modify another bunch of code which is out of the context of this PR (refer 2.3.2 in BEP#1).

Reply:
Ideally:

```
if is_valid_transaction(tx_dict):
    transaction = Transaction.from_dict(tx)

if transaction:
   ...
```

But I see the mess around it now so I am fine with keeping it as is for now, sorry for holding that off for a little while.

Reply:
@ldmberman No problem, I am happy to answer the queries.